rfc/src/rfcs/0003-cover-statement.md

```diff
@@ -0,0 +1,141 @@
+- **Feature Name:** Cover statement `cover_statement`
+- **Feature Request Issue:** <https://github.com/model-checking/kani/issues/696>
+- **RFC PR:** <https://github.com/model-checking/kani/pull/1906>
+- **Status:** Under Review
+- **Version:** 0
+- **Proof-of-concept:** *Optional field. If you have implemented a proof of concept, add a link here*
+
+## Summary
+
+A new Kani API that allows users to check that a certain condition can occur at a specific location in the code.
+
+## User Impact
+
+Users typically want to gain confidence that a proof checks what it is supposed to check, i.e. that properties are not passing vacuously due to an over-constrained environment.
+
+A new Kani macro, `cover` will be created that can be used for checking that a certain condition _can_ occur at a specific location in the code.
+The purpose of the macro is to verify, for example, that assumptions are not ruling out those conditions, e.g.:
+```rust
+let mut v: Vec<i32> = Vec::new();
+let len: usize = kani::any();
+kani::assume(len < 5);
+for _i in 0..len {
+    v.push(kani::any());
+}
+// make sure we can get a vector of length 5
+kani::cover!(v.len() == 5);
+```
+This is typically used to ensure that verified checks are not passing _vacuously_, e.g. due to overconstrained pre-conditions.
+
+The special case of verifying that a certain line of code is reachable can be achieved using `kani::cover!()` (which is equivalent to `cover!(true)`), e.g.
+```rust
+match x {
+    val_1 => ...,
+    val_2 => ...,
+    ...
+    val_i => kani::cover!(), // verify that `x` can take the value `val_i`
+}
+```
+
+Similar to Rust's `assert` macro, a custom message can be specified, e.g.
+```rust
+kani::cover!(x > y, "x can be greater than y");
+```
+
+## User Experience
+
+The `cover` macro instructs Kani to find _at least one_ possible execution that satisfies the specified condition at that line of code.  If no such execution is possible, the check is reported as *unsatisfiable*.
+
+Each cover statements will be reported as a check whose description is `cover condition "cond"` and whose status is:
+- `SATISFIED` (green): if Kani found an execution that satisfies the condition
+- `UNSATISFIABLE` (yellow): if Kani proved that the condition cannot be satisfied
+- `UNREACHABLE` (yellow): if Kani proved that the cover statement itself cannot be reached
```

Comment:
Just a few suggestions in this paragraph:
 1. "statements" (1st occurrence) -> "statement"
 2. Dots at the end of each sentence.
 3. Maybe use ``cover condition `<cond>` ``?

Reply:
Fixed the first two.

For the third, I changed it to:
```
cover condition: cond
```
to mimic the assert message:
```
assertion failed: 1 + 1 == 3
```
I hope this is OK.